app/server/api/vendor_payout_api.py

```diff
@@ -0,0 +1,177 @@
+                v.balance,
+                v.total_sent,
+                v.total_received,
+                v.is_approved,
+                v.is_beneficiary,
+                v.is_vendor,
+                transfer.id,
+                v.balance,
+                '',
+                '',
+            ])
+        # Encode the CSV such that it can be sent as a file
+        bytes_output = io.BytesIO()
+        bytes_output.write(output.getvalue().encode('utf-8'))
+        bytes_output.seek(0)
+        return send_file(bytes_output, as_attachment=True, attachment_filename='vendor_payout.csv', mimetype='text/csv')
+
+class ProcessVendorPayout(MethodView):
+    @requires_auth(allowed_roles={'ADMIN': 'sempoadmin'})
+    def post(self):
+        # Handle a file upload, or CSV in JSON
+        if request.files:
+            flask_file = request.files['file']
+            stream = codecs.iterdecode(flask_file.stream, 'utf-8')
+            reader = csv.DictReader(stream)
+        else:
+            post_data = request.get_json()
+            if not post_data:
+                raise Exception('Please provide a CSV file')
+            csv_data = post_data.get('csv_data', [])
+            f = io.StringIO(csv_data)
+            reader = csv.DictReader(f)
+        transfers = []
+        for line in reader:
+            vendor = db.session.query(TransferAccount).filter(TransferAccount.id == line['ID']).first()
+            if not vendor.is_vendor:
+                raise Exception(f'{vendor} is not a vendor!')
+            transfer = db.session.query(CreditTransfer).filter(CreditTransfer.id == line['Transaction ID']).first()
+            if not transfer:
+                raise Exception(f'Tranfer with ID {line["Transaction ID"]} not found!')
+            if float(transfer.transfer_amount) != float(line["Amount Due Today"]):
+                raise Exception(f'Invalid transfer amount!')
+            if line['Payment Has Been Made'].upper() == 'TRUE' and line['Bank Payment Date']:
+                try:
+                    transfer.resolve_as_complete_and_trigger_blockchain()
+                except:
+                    pass
+            elif line['Payment Has Been Made'] == 'FALSE':
+                transfer.resolve_as_rejected()
+            transfers.append(transfer)
+
+        output = io.StringIO()
+        writer = csv.writer(output)
+
+        writer.writerow([
+            'ID', 
+            'First Name', 
+            'Last Name', 
+            'Transfer Created',
+            'Transfer Type',
+            'Transfer Amount',
+            'Transfer Status',
+        ])
+        for t in transfers:
+            writer.writerow([
+                t.sender_transfer_account.id,
+                t.sender_transfer_account.primary_user.first_name,
+                t.sender_transfer_account.primary_user.last_name,
+                t.created,
+                t.transfer_type,
+                t.transfer_amount,
+                t.transfer_status
```

Comment:
ensure enums display values

Reply:
`t.transfer_type.value` should work

Reply:
Just used enums to reflect the [doc](https://docs.google.com/spreadsheets/d/1RRbaH3wLlVaiUnVWQV5wP3E0YGRUIp7j/edit#gid=1305560977). Made the change though!

Reply:
Ah yes, that makes sense. Thanks!